packages/cli/src/ceramic-daemon.ts

```diff
@@ -237,6 +238,21 @@ class CeramicDaemon {
     next()
   }
 
+  /**
+   * Create read-only document from genesis record
+   */
+  async createReadOnlyDocFromGenesis (req: Request, res: Response, next: NextFunction): Promise<void> {
```

Comment:
maybe call this `loadDocumentFromGenesis`?

Reply:
I was thinking that expect we already have `ceramic.loadDocument` which takes in a docId and internally does `Document.load`. So it might be confusing to use the name load here because it internall maps to `ceramic.createDocumentFromGenesis` and `Document.create`

Reply:
Yeah, the fact that it uses different functions internally doesn't bother me, as that's an implementation detail.  What bothers me slightly more is that it doesn't return `null` if the doc doesn't already exist, like loadDocument does.  I wish we had a client-side way to get the CID for some content without having to *actually* publish that content to IPFS.  I wonder if we could find the hashing code in IPFS and expose it if it isn't already.  Anyway that's probably not worth exploring right now.

I think behavior-wise what you wrote here is fine for now, the only question is the name.  I'd probably lean towards something like `loadDocumentFromGenesis` or `loadDocFromGenesisContent`, but add a TODO comment that it shouldn't publish the genesis record to IPFS if it isn't already there.  But I don't feel strongly here

Reply:
Will push this through for now but you are definitely right--the current functionality does `ipfs.dag.put` so it technically is creating something.

For that reason I'd rather the function name be more accurate so we know what this is doing (if it were public facing like the API I'd agree with you but this is hidden behind `/documents` anyway).

I'll create an additional story and add a TODO to modify this behavior